### PR TITLE
Initial cleanup

### LINKED ISSUE #19: Initial cleanup
Complete a basic cleanup of smaller issues before addressing the rest of the codebase.

Tasks:

* [X] Fix imports
* [X] Basic PEP8 cleanup
* [X] Fix obvious warnings
* [X] Fix variable shadowing

### PR DESCRIPTION
**Pull request by truth-quark**
Resolves #19. This is an initial code cleanup to clean up formatting/PEP8 issues, warnings & some minor issues before starting the main testing retrofit. Unfortunately it's a larger commit than I'd like...

I've manually tested this to ensure the outputs from this version are the same as Martin's original version.

For the review, any 2nd opinions on the partial restructure, gaps & initial correctness checks are welcome...

### CODE REVIEW

**Review by blimlim — APPROVED** [review]
Everything looks ok to me!

File: src/um2netcdf.py [hunk 1]
```diff
@@ -347,52 +403,57 @@ def keyfunc(c):
...
             cubewrite(c, sman, args.compression, args.use64bit, args.verbose)
 
+
 if __name__ == '__main__':
-    import sys, argparse
     parser = argparse.ArgumentParser(description="Convert UM fieldsfile to netcdf")
     parser.add_argument('-k', dest='nckind', required=False, type=int,
-                        default=3, help='specify kind of netCDF format for output file: 1 classic, 2 64-bit offset, 3 netCDF-4, 4 netCDF-4 classic model. Default 3', choices=[1,2,3,4])
+                        default=3,
+                        help='specify netCDF output format: 1 classic, 2 64-bit offset, 3 netCDF-4, 4 netCDF-4 classic model. Default 3',
```
Comment (jo-basevi) [inline src/um2netcdf.py:434]: I think this line will get PEP8 errors for line too long

**Review by jo-basevi — APPROVED** [review]
Changes look good to me! I just noticed one small pep8 error. Do you use `black`(https://pypi.org/project/black/) or `ruff` for code formatting? It could be useful for it to be run over the code to help with the formatting.

I saw there are a couple TODOs added in code for renaming variables. Could be worth eventually renaming a couple to use `snake_case`, e.g. `newdate` to `new_date`

**Comment by truth-quark** [comment]
> Changes look good to me! I just noticed one small pep8 error. Do you use `black`(https://pypi.org/project/black/) or `ruff` for code formatting? It could be useful for it to be run over the code to help with the formatting.

I've used `black`. Adding a code formatter is on the cards for another issue & PR. I'll add that as a new issue.
 
> I saw there are a couple TODOs added in code for renaming variables. Could be worth eventually renaming a couple to use `snake_case`, e.g. `newdate` to `new_date`

Thanks, I'll flag that for a separate renaming/refactoring/readability issue.

File: src/um2netcdf.py (hunk 1 above)
Comment (truth-quark) [reply src/um2netcdf.py:434]: Yep, I had the error but decided to defer string fixes to another issue/PR.

**Pull request merged**